orchestra/utils/task_lifecycle.py

```diff
@@ -275,6 +228,182 @@ def assign_task(worker_id, task_id):
     return task
 
 
+@transaction.atomic
+def reassign_assignment(worker_id, assignment_id):
+    worker = Worker.objects.get(id=worker_id)
+    assignment = TaskAssignment.objects.get(id=assignment_id)
+    if assignment.assignment_counter > 0:
+        role = WorkerCertification.Role.REVIEWER
+    else:
+        role = WorkerCertification.Role.ENTRY_LEVEL
+
+    if not _worker_certified_for_task(worker, assignment.task, role):
+        raise WorkerCertificationError('Worker not certified for this task.')
+    if is_worker_assigned_to_task(worker, assignment.task):
+        raise TaskAssignmentError('Worker already assigned to this task.')
+
+    assignment.worker = worker
+    assignment.save()
+    add_worker_to_project_team(worker, assignment.task.project)
+
+    return assignment
+
+
+@transaction.atomic
+def complete_and_skip_task(task_id):
+    task = Task.objects.get(id=task_id)
+    task.status = Task.Status.COMPLETE
+    task.save()
+    for assignment in task.assignments.all():
+        assignment.status = TaskAssignment.Status.SUBMITTED
+        assignment.save()
+    create_subsequent_tasks(task.project)
+    return task
+
+
+@transaction.atomic
+def revert_task_to_datetime(task_id, revert_datetime, fake=False):
+    task = Task.objects.get(id=task_id)
+    audit = {'task': task, 'change': None, 'assignments': []}
+
+    assignments_after = (assignment_history(task)
+                         .filter(start_datetime__gt=revert_datetime)
+                         .all())
+
+    if assignments_after.exists():
+        for assignment in assignments_after:
+            audit['assignments'].append({'assignment': assignment,
+                                         'change': 'deleted',
+                                         'snapshots': []})
+        if not fake:
+            # Delete assignments created after the revert datetime.
+            assignments_after.delete()
+
+    for assignment in assignment_history(task).all():
+        assignment_audit = _revert_assignment_to_datetime(
+            assignment, revert_datetime, fake)
+        if assignment_audit:
+            audit['assignments'].append(assignment_audit)
+
+    # Delete task if it starts after revert_datetime
+    if task.start_datetime > revert_datetime:
+        if not fake:
+            task.delete()
+        audit['change'] = 'deleted'
+    elif audit['assignments']:
+        if not fake:
+            _revert_task_status(task)
+        audit['change'] = 'reverted'
+
+    return audit
+
+
+@transaction.atomic
```

Comment:
do you actually want this to be recursively atomic inside of the other revert?  if not, remove.  you typically do this when you want to be able to partially roll back (in the database transaction sense) a transaction